chainer/functions/math/tensordot.py

```diff
@@ -0,0 +1,175 @@
+import collections
+import numpy
+import six
+
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+
+def _tensordot(a, b, a_axes, b_axes, c_axes=None):
+
+    a_col_ndim = len(a_axes[1])
+    b_row_ndim = len(b_axes[0])
+    if a_col_ndim != b_row_ndim:
+        raise ValueError('axes count mismatch')
+    if a.ndim < a_col_ndim or b.ndim < b_row_ndim:
+        raise ValueError('dimension of input tensors must be '
+                         'greater equal to dot-axes count ({})'
+                         .format(a_col_ndim))
+    for a_axis, b_axis in zip(a_axes[1], b_axes[0]):
+        if a.shape[a_axis] != b.shape[b_axis]:
+            raise ValueError('shape mismatch')
+
+    if a.ndim == 0 or b.ndim == 0:
+        y = a * b
```

Comment:
Is this for performance reasons or does `xp.tensordot` in this case not work?

Reply:
I thought `xp.tensordot` did not work when ndim=0, but seems it was because `list` was used for `axes` option at that time. It works OK if tuple is used. will use `xp.tensordot` even when ndim=0.